File: frontend/src/app/models/HeureSup.ts
import { Employe } from "./Employe";
import { Tarif } from "./Tarif";

export class HeureSup{

    public id?:number;

    public date!:Date;

    public nb_heures!:number;

    public employe!:Employe;

    public tarif!:Tarif;
    
}

File: backend/src/main/java/tn/devops/backend/entities/HeureSup.java
package tn.devops.backend.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;


@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class HeureSup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private Date date;

    private double nb_heures;

    @ManyToOne(cascade = CascadeType.ALL)
    private Employe employe;

    @ManyToOne(cascade = CascadeType.ALL)
    private Tarif tarif;
}
